### PR TITLE
feat: support blob responses

### LINKED ISSUE #385: fix sendProxy function
### Environment

nuxt 3 with nitro 2.4.0, node v18.14.2

### Reproduction

https://stackblitz.com/edit/github-xzp2dt?file=nuxt.config.ts,app.vue

### Describe the bug

It seems sendProxy function doesn't respond with correct data

I investigated this issue here: https://github.com/unjs/nitro/issues/1237

when loading images, 304 response and blob data are not handled correctly

this is the cause: https://github.com/unjs/h3/blob/2cbcb7d4bd5a2c50d316de5b23ed7124bf0bfe68/src/utils/proxy.ts#L109

### Additional context

I managed to fix it locally by using this change (probably not a good solution, just wanted to make it work):

```js
if (response._data !== undefined) {
    if (response._data && response._data.arrayBuffer) {
      await response._data.arrayBuffer().then(b => {
        const buffer = Buffer.from(b);
        event.node.res.write(buffer);
      });
    }
    return event.node.res.end();
  }
```

edit: updated solution (took from the code below it):
```js
if (response._data !== undefined) {
  if (response._data && response._data.arrayBuffer) {
      const data = new Uint8Array(await response._data.arrayBuffer());
      return event.node.res.end(data);
    }
    return response._data;
 }
```

### Logs

_No response_

### PR DESCRIPTION
**Pull request by jd1378**
Resolves #385 and https://github.com/unjs/nitro/issues/1237

### CODE REVIEW

**Comment by pi0** [comment]
Would you please add a unit test? 🙏🏼

**Comment by jd1378** [comment]
Well during writing tests I realized the real issue was that Blob type was not being handled properly, which was why the fix was working, So I handled it in `app` part
but still I saw when empty string is returned along with 304, it has problem handling it and responds with 500, so I still added the test and fix for that part as well

**Comment by jd1378** [comment]
I have pushed a commit to fix the lint issues but I don't the commit in the PR, what am I doing wrong ? is it me or github being funny ?

**Comment by jd1378** [comment]
yeah it was GitHub, re-set the base branch on edit dialog and it showed up

**Comment by Hebilicious** [comment]
@jd1378 This is a  very good change, thanks for adding the tests !

File: test/proxy.test.ts [hunk 1]
```diff
@@ -106,6 +107,36 @@ describe("", () => {
...
+      app.use(
+        "/",
+        eventHandler((event) => {
+          return proxyRequest(event, url + "/reply-empty-304", { fetch });
+        })
+      );
+
+      let result: Response | null;
+
+      try {
```
[the remark this answers is not in the document]
Comment (jd1378) [reply test/proxy.test.ts:129]: Well I'm not sure, if throwing inside the test is not causing any problem (other than failing, obviously), then probably we can  safely remove it

File: src/utils/proxy.ts [hunk 2]
```diff
@@ -133,9 +133,11 @@ export async function sendProxy(
     return event.node.res.end(data);
   }
 
-  // Send as stream
-  for await (const chunk of response.body as any as AsyncIterable<Uint8Array>) {
-    event.node.res.write(chunk);
+  if (response.body !== null) {
```
[the remark this answers is not in the document]
Comment (jd1378) [reply src/utils/proxy.ts:136]: yeah, but I don't think body can be falsy other than null, which is why I checked it, but again just to be safe we can check in general
I'll push a commit soon

**Comment by jd1378** [comment]
pushed commit for both suggestions

**Review by Hebilicious — APPROVED** [review]

**Pull request merged**